javalib/src/main/scala/java/util/concurrent/atomic/AtomicBoolean.scala

```diff
@@ -1,16 +1,34 @@
 package java.util.concurrent.atomic
 
+import scala.scalanative.native
+import scala.scalanative.native.Atomic
+
 class AtomicBoolean(private[this] var value: Boolean) extends Serializable {
   def this() = this(false)
 
+  private val lock: native.Ptr[native.CInt] = native.stackalloc[native.CInt]
```

Comment:
Would it make sense to either use the C atomic API, or go the route of implementing sun.misc.Unsafe using LLVMs atomic primitives and using as a base uthe JVM atomic API?